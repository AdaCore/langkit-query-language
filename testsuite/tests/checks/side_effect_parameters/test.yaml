driver: 'checker'
rule_name: side_effect_parameters
project: prj.gpr
rule_arguments:
   side_effect_parameters.functions: '["p1.read","p1.read2"]'
