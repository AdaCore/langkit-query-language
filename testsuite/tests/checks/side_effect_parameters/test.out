p1.adb:7:13: rule violation: call with side effects
7 |       Proc (Read (1), Read (2)); --  FLAG
  |             ^^^^^^^^^^^^^^^^^^

p1.adb:9:13: rule violation: call with side effects
9 |       Proc (Read, Read);         --  FLAG
  |             ^^^^^^^^^^

p1.adb:10:13: rule violation: call with side effects
10 |       Proc (Read2, Read2);       --  FLAG
   |             ^^^^^^^^^^^^

p1.adb:12:13: rule violation: call with side effects
12 |       Proc (Func1 (Read), Read); --  FLAG
   |             ^^^^^^^^^^^^^^^^^^

p1.adb:13:13: rule violation: call with side effects
13 |       Proc (Read, Read_R);       --  FLAG
   |             ^^^^^^^^^^^^

